src/PostgREST/App.hs

```diff
@@ -322,7 +329,7 @@ app dbStructure proc conf apiRequest =
       rangeHeader queryTotal tableTotal =
         let lower = rangeOffset topLevelRange
             upper = lower + toInteger queryTotal - 1
-            contentRange = contentRangeH lower upper (toInteger <$> tableTotal)
+            contentRange = contentRangeH (isJust tableTotal) lower upper $ (toInteger . fromJust) tableTotal
```

Comment:
We should [avoid partial functions](https://wiki.haskell.org/Avoiding_partial_functions) as much as we can. There are some of those in the codebase, but I've been removing them in any chance I get. Could you express this in other way and avoid `fromJust`?

Reply:
Yes, I understand.

I've changed to this form, because it was shorter and `total` was never instantiated if `shouldCount` was false. 

Anyway, to avoid this form, which is I agree, is not ideal, I just reverted to the previous form.